### DISCUSSION TITLE
Video Capture Source - Rotation Data

### DISCUSSION THREAD

**Discussion by bytowneboy** [WarmUpTill/SceneSwitcher#648] [Ideas]
I occasionally do iPhone support for clients with a USB capture card, and have different scenes for portrait vs landscape orientation for the phone. 
I'm optimistic I can wizard up something with the JSON properties, but it feels like it would be a good addition to the macro builder.
Thanks for your amazing work!

**Comment by WarmUpTill** [comment]
Thanks for the suggestion!

Can you give more details how you imagine this functionality to be incorporated into the plugin?
I do not fully understand yet how OBS reacts to the change - does the source provided by the capture card dynamically change teh source's resolution?

**Reply by bytowneboy** [reply]
![Screenshot 2022-12-31 at 12 07 35](https://user-images.githubusercontent.com/121645918/210150781-c27f09e5-5ae4-4e69-84e2-f6099024682a.png)
![Screenshot 2022-12-31 at 12 07 47](https://user-images.githubusercontent.com/121645918/210150782-7ae597a4-073f-4e63-b37d-c7dda65f33c7.png)
I think that the video source is sending "rotation data" into OBS, while it doesn't seem to change the resolution, it reorients the source from portrait to landscape--there is a chance that this is being handled by my capture card. The reason I have two different scenes is that I have the two orientations cropped very differently in my scenes.
As far as integration, maybe rotation state could be included in the 'if source settings match' (see attachment).
With apologies if my explanation or request are a little incoherent, I don't have a very deep understanding of how much of this works.

I'm hoping that the 'apply rotation data from the camera, if any' referenced by the video capture source dialogue is present, applicable to my use case and available for integration: https://obsproject.com/kb/video-capture-sources 

![Screenshot 2022-12-31 at 11 51 05](https://user-images.githubusercontent.com/121645918/210150513-217bec5f-ba78-4592-a193-e3406dc76f41.png)

**Reply by WarmUpTill** [reply]
Unfortunately it seems like there is no API call to read the rotation data (or I am missing something).
There is `obs_source_set_async_rotation` but no corresponding getter function.

I will try to see if there maybe is another way to access this information when I find the time.
